fastlane/lib/fastlane/actions/notarize.rb

```diff
@@ -123,6 +130,35 @@ def self.staple(package_path, verbose)
         )
       end
 
+      def self.with_notarize_authenticator(api_key_path)
+        if api_key_path
+          # From xcrun altool for --apiKey:
+          # This option will search the following directories in sequence for a private key file with the name of 'AuthKey_<api_key>.p8':  './private_keys', '~/private_keys', '~/.private_keys', and '~/.appstoreconnect/private_keys'.
+          api_key = JSON.parse(File.read(api_key_path))
+          api_key_folder_path = '~/.appstoreconnect/private_keys'
+          api_key_file_path = File.join(api_key_folder_path, "AuthKey_#{api_key['key_id']}.p8")
+          directory_exists = File.directory?(api_key_folder_path)
+          file_exists = File.exist?(api_key_file_path)
+          begin
+            FileUtils.mkdir_p(api_key_folder_path) unless directory_exists
+            File.open(api_key_file_path, 'w') { |f| f.write(api_key['key']) } unless file_exists
+
+            yield(proc { |command| "#{command} --apiKey #{api_key['key_id']} --apiIssuer #{api_key['issuer_id']}" })
```

Comment:
I’m thinking maybe we should do some validation `key` and `key_id` exist in `api_key` at the start of this method to make any debugging easier on the user if something isn’t right?

Reply:
or maybe we just reuse this 
https://github.com/fastlane/fastlane/blob/6702e2eeef1f383b3627b940f6e6c76d4ebc1551/spaceship/lib/spaceship/connect_api/token.rb#L28-L40